logstash-core/src/main/java/org/logstash/FileLockFactory.java

```diff
@@ -39,7 +41,8 @@
 
     private FileLockFactory() {}
 
-    private static final Set<String> LOCK_HELD = Collections.synchronizedSet(new HashSet<String>());
+    private static final Set<String> LOCK_HELD = Collections.synchronizedSet(new HashSet<>());
```

Comment:
No, you could use FileLock objects independently of the `FileLockFactory`.

I could have made that more obvious by using a Lock class wrapping the FileLock but I thought it was pretty much unnecessary overhead.